src/VolumeDrawable.ts

```diff
@@ -685,4 +696,75 @@ export default class VolumeDrawable {
     this.volumeRendering.setRotation(this.rotation);
     this.meshVolume.setRotation(this.rotation);
   }
+
+  setupGui(pane: Pane, onChange?: () => void): void {
+    pane.addInput(this, "translation").on("change", ({ value }) => this.setTranslation(value));
+    pane.addInput(this, "rotation").on("change", ({ value }) => this.setRotation(value));
+
+    const pathtrace = pane.addFolder({ title: "Pathtrace", expanded: false });
+    pathtrace.addInput(this, "primaryRayStepSize").on("change", ({ value }) => this.setRayStepSizes(value));
+    pathtrace
+      .addInput(this, "secondaryRayStepSize")
+      .on("change", ({ value }) => this.setRayStepSizes(undefined, value));
+
+    const channels = pane.addFolder({ title: "Channels", expanded: false });
+    this.channelOptions.forEach((options, channelIndex) => {
+      // eslint-disable-next-line @typescript-eslint/no-this-alias
+      const self = this;
+      type ChannelState = WithObjectColors<VolumeChannelDisplayOptions>;
+      // Volume channel options are not stored in a consistent place.
+      // Perhaps they should be. For now, this proxy lets us pretend they are.
+      const optionsProxy = new Proxy(options as ChannelState, {
```

Comment:
along the lines of my other comment, I would have just made a simple data structure to hold the "tweakable" values per channel and then map the change handlers of the tweakpane to drill in and set the stuff.  But this is very instructive to me about how to use Proxy!

Reply:
Yeah, I don't necessarily love what I came up with here, but I wanted to avoid creating new values which could potentially become out of sync with the settings used for rendering. (Though these would still have to be deliberately propagated to tweakpane with `refresh`.)

Despite the effort it took to work out, I wouldn't mind if it didn't make it into the merged version of this PR. The main value of it for me was learning the providence of all the settings in it, and it will remain in the commit graph should it ever be useful to reference later.